api/src/security/auth-utils.ts

```diff
@@ -215,6 +215,10 @@ export const userHasValidSystemRoles = function (
     userSystemRoles = [userSystemRoles];
   }
 
+  if (!validSystemRoles.length) {
+    return true;
+  }
+
```

Comment:
I ended up adding a check at the top of the function (206-208), so this shouldn't be needed anymore.

Having the check here caused a funny issue where some falsy values would still get added to the array, and then the length wouldn't be 0, even though it didn't really contain a role.

Reply:
Noticed during unit testing :)